x/stakeibc/keeper/keeper_test.go

```diff
@@ -44,3 +45,28 @@ func (s *KeeperTestSuite) GetMsgServer() types.MsgServer {
 func TestKeeperTestSuite(t *testing.T) {
 	suite.Run(t, new(KeeperTestSuite))
 }
+
+func (s *KeeperTestSuite) TestIsRedemptionRateWithinSafetyBounds() {
+	params := s.App.StakeibcKeeper.GetParams(s.Ctx)
+	params.MinRedemptionRates["osmosis-1"] = "0.3"
+	params.MaxRedemptionRates["osmosis-1"] = "2.0"
+	s.App.StakeibcKeeper.SetParams(s.Ctx, params)
+
+	zone := types.HostZone{
+		ChainId:        "osmosis-1",
+		RedemptionRate: sdk.NewDec(1),
+	}
+	rrSafe, err := s.App.StakeibcKeeper.IsRedemptionRateWithinSafetyBounds(s.Ctx, zone)
+	s.Require().NoError(err)
+	s.Require().True(rrSafe)
+
+	zone.RedemptionRate = sdk.NewDecWithPrec(16, 1) // 1.6
+	rrSafe, err = s.App.StakeibcKeeper.IsRedemptionRateWithinSafetyBounds(s.Ctx, zone)
+	s.Require().NoError(err)
+	s.Require().True(rrSafe)
+
+	zone.RedemptionRate = sdk.NewDecWithPrec(21, 1) // 2.1
```

Comment:
Could you add a test for a zone that does NOT have a custom bound, to verify that in that case it uses the default bounds?

Reply:
Can you also add another host zone to the params and add some tests across both hosts? 
E.g. Osmo 0.3-2, Gaia 1.5-2.5
Test cases:
 0.1 - out of both
0.3 - in for osmo, out for gaia
1.5 - in for both
2.2 - out for osmo, in for gaia

Reply:
Okay got it.

Reply:
https://github.com/Stride-Labs/stride/pull/614/commits/85508ca1f81e4d93780cf6a82585e5f0a84d3c1c

Reply:
beautiful!